# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: memory/build/mozjemalloc.h
# Commit: f9e3f8796616
# Full Hash: f9e3f8796616bf24fd9d1c7260bc126fe531d0e0
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2025-02-19 21:53:46
# Regressor Bug: 1920451
# File Overlap Count: 1
# Description:
#   Bug 1920451 - Introduce a concept of last significant re-use to decide when to purge. r=pbone,smaug
#   
#   We want to better predict when it is a good moment to purge. D220616 answered well the question, when to best process a purge when it is needed (that is, during main thread idle time), but that does not necessarily mean it is a good idea for a single arena, to purge in that moment as arenas can be used on other threads intensely while the main thread is idle.
#   
#   If we saw a recent re-use in an arena we can most likely expect to come more of that and wait with purge a little more. So whenever we are able to re-use a dirty page, we assume that this is a sign of a significant re-use that avoided an allocation of new memory.
# ==============================================================================

diff -r 4eeb6f53b113 -r f9e3f8796616 memory/build/mozjemalloc.h
--- a/memory/build/mozjemalloc.h	Wed Feb 19 07:07:49 2025 +0000
+++ b/memory/build/mozjemalloc.h	Wed Feb 19 07:21:15 2025 +0000
@@ -171,7 +171,10 @@
 
   static bool moz_enable_deferred_purge(bool aEnable) { return false; }
 
-  static bool moz_may_purge_one_now(bool aPeekOnly) { return false; }
+  static purge_result_t moz_may_purge_one_now(bool aPeekOnly,
+                                              uint32_t aReuseGraceMS) {
+    return purge_result_t::Done;
+  }
 
 #define MALLOC_DECL(name, return_type, ...)                 \
   static return_type moz_arena_##name(                      \